ksql-rest-app/src/test/java/io/confluent/ksql/rest/server/resources/KsqlResourceTest.java

```diff
@@ -970,6 +978,56 @@ public void shouldNotIncludeSslPropertiesInListPropertiesOutput() {
   }
 
   @Test
+  public void shoudlHandleTerminateRequestCorrectly() throws InterruptedException {
+    // Given:
+    final KsqlResource testResource = getMockKsqlResource();
+    final ClusterTerminateRequest request = new ClusterTerminateRequest(ImmutableList.of("Foo"));
+
+    // When:
+    final Response response = testResource.terminateCluster(request);
+
+    // Then:
+    assertThat(response.getStatus(), equalTo(200));
+  }
+
+  @Test
+  public void shoudlFailIfCannotWriteTerminateCommand() throws InterruptedException {
+    // Given:
+    final KsqlResource testResource = getMockKsqlResource();
+    final ClusterTerminateRequest request = new ClusterTerminateRequest(ImmutableList.of("Foo"));
+    when(replayableCommandQueue.enqueueCommand(any(), any(), any(), any()))
+        .thenThrow(new RuntimeException());
+
+    // When:
+    final Response response = testResource.terminateCluster(request);
+
+    // Then:
+    assertThat(response.getStatus(), equalTo(500));
+    assertThat(response.getEntity().toString(),
+        startsWith("Could not write the statement 'TERMINATE CLUSTER;' into the command "));
+  }
+
+  @Test
+  public void shouldFailForInvalidTerminateClusterParameters() throws InterruptedException {
+    // Given:
+    final KsqlResource testResource = getMockKsqlResource();
+    final ClusterTerminateRequest request = new ClusterTerminateRequest(
+        ImmutableList.of("[Invalid Regex"));
+
+    // When:
+    try {
+      testResource.terminateCluster(request);
+      fail();
+    } catch (final Exception e) {
+      assertThat(e, instanceOf(KsqlRestException.class));
```

Comment:
It's cleaner to do this kind of thing using `ExpectedException`. Though it does require custom matchers. But said matchers are reusable.

As an example, I've added matchers to do this in this PR: https://github.com/confluentinc/ksql/pull/2273

Let's get that merged and then you can use them in this test :D